tutorials/tour/_posts/2017-02-13-unified-types.md

```diff
@@ -48,3 +48,15 @@ c
 true
 <function>
 ```
+
+## Nothing and Null
+`Nothing` is a subtype of all types. There is no value that has type `Nothing`. There are two uses: The first is to signal abnormal termination such as an exception (i.e. it doesn't return). The second type usage is an element type of empty collections (e.g. `Set[Nothing]`).
+
```

Comment:
I think we could say something like "we can conclude from the type `List[Nothing]` that  `Nil` must be empty" which gets the point across while glossing over variance. This is also the case for `Set[Nothing]` of course, but as you noted it ends up not being useful due to invariance.